.github/workflows/builder_docker-based_slsa3.yml

```diff
@@ -143,90 +153,118 @@ jobs:
+          path: "${{ env.BUILDER_BINARY }}"
+          sha256: "${{ needs.generate-builder.outputs.builder-binary-sha256 }}"
+          set-executable: true
+
+      - name: Get source ref
+        id: resolve
+        env:
+          SOURCE_DIGEST: ${{ inputs.source-digest }}
+          GITHUB_SHA: ${{ github.sha }}
+        run: |
+          if [ -z "$SOURCE_DIGEST" ]; then 
+            ref=$(echo ${SOURCE_DIGEST} | cut -d':' -f2)
+            echo "ref=$ref" >> $GITHUB_OUTPUT
+          else 
+            echo "ref=$GITHUB_SHA" >> $GITHUB_OUTPUT
+          fi
+
+      # TODO: Use secure-checkout, but this requires configuring the source-repository.
+      - name: Checkout the source repository
+        uses: actions/checkout@755da8c3cf115ac066823e79a1e1788f8940201b # v3.2.0
+        with:
+          fetch-depth: 1
+          persist-credentials: false
+          path: __PROJECT_CHECKOUT_DIR__
+          repository: ${{ inputs.source-repository || github.repository }}
+          ref: ${{ steps.resolve.outputs.ref }}
+
       - name: Generate build definition
         id: generate
         # These are the inputs, it may be with: for an action or
         # specified with these env vars.
         env:
           BUILDER_IMAGE: ${{ inputs.builder-image }}
           BUILDER_DIGEST: ${{ inputs.builder-digest }}
-          BUILDER_VERIFIED: ${{ needs.verify-builder-image.outputs.verified }}
           CONFIG_PATH: ${{ inputs.config-path }}
-          OUTPUT: ${{ inputs.builder-output-path }}
           # Either the provided source repository or the caller's GITHUB_REPOSITORY
           SOURCE_REPOSITORY: ${{ inputs.source-repository || github.repository }}
           # Either the provided digest, or the caller's GITHUB_REF
           SOURCE_DIGEST: ${{ inputs.source-digest || github.ref }}
         run: |
-          # Run builder binary and produce buildDefinition
-          # TODO: Insert verified builder image provenance into resolvedDep
-          echo "run resolution and output buildDefinition!"
+          set -euo pipefail
+
+          # Note: this outputs information about resolved arguments, etc.
+          # the values are trusted because the compiler is not invoked.
+          echo "$GITHUB_WORKSPACE/$BUILDER_BINARY" dry-run \
+            --build-config-path ${CONFIG_PATH} \
+            --build-definition-path build-definition.json \
+            --builder-image ${BUILDER_IMAGE}@${BUILDER_DIGEST} \
+            --git-commit-hash ${SOURCE_DIGEST} \
+            --source-repo git+https://github.com/${SOURCE_REPOSITORY}
+          "$GITHUB_WORKSPACE/$BUILDER_BINARY" dry-run \
+            --build-config-path ${CONFIG_PATH} \
+            --build-definition-path build-definition.json \
+            --builder-image ${BUILDER_IMAGE}@${BUILDER_DIGEST} \
+            --git-commit-hash ${SOURCE_DIGEST} \
+            --source-repo git+https://github.com/${SOURCE_REPOSITORY}
```

Comment:
```suggestion
          # Note: this outputs information about resolved arguments, etc.
          # the values are trusted because the compiler is not invoked.
          echo "${GITHUB_WORKSPACE}/${BUILDER_BINARY}" dry-run \
            --build-config-path "${CONFIG_PATH}" \
            --build-definition-path build-definition.json \
            --builder-image "${BUILDER_IMAGE}@${BUILDER_DIGEST}" \
            --git-commit-hash "${SOURCE_DIGEST}" \
            --source-repo "git+https://github.com/${SOURCE_REPOSITORY}"
          "${GITHUB_WORKSPACE}/${BUILDER_BINARY}" dry-run \
            --build-config-path "${CONFIG_PATH}" \
            --build-definition-path build-definition.json \
            --builder-image "${BUILDER_IMAGE}@${BUILDER_DIGEST}" \
            --git-commit-hash "${SOURCE_DIGEST}" \
            --source-repo "git+https://github.com/${SOURCE_REPOSITORY}"
```